pandas/io/formats/style.py

```diff
@@ -3133,8 +3144,16 @@ def highlight_null(
         def f(data: DataFrame, props: str) -> np.ndarray:
             return np.where(pd.isna(data).to_numpy(), props, "")
 
+        if null_color != "red":
```

Comment:
you need to make `null_color` == `lib.no_default` here (and then if its not that you can warn)

Reply:
done